### PR TITLE
Initial configuration for react testing library

### PR DESCRIPTION
**Pull request by gildub**
- Adds @testing-library/user-event package
- Moves jest.config.js to `./config
- Sets off coverage flag by default in jest.config.js so it's used only when needed by activating the option instead of having to turning off.

Resolves https://github.com/konveyor/forklift-ui/issues/320

### CODE REVIEW

**Comment by mturley** [comment]
Huh, the test that is complaining about JSX being invalid ran fine before these config changes. I wonder what that's all about.

**Comment by gildub** [comment]
@mturley, yes sorry, it was missing a couple of things.

File: package.json [hunk 1]
```diff
@@ -18,10 +18,12 @@
     "start:dev": "./scripts/start-dev.sh",
     "start:dev:remote": "npm run setup:dev:remote && DATA_SOURCE=remote npm run start:dev",
     "start:dev:mock": "DATA_SOURCE=mock npm run start:dev",
-    "test": "$(npm bin)/jest --coverage",
     "test:watch": "npm run test -- --watch",
```
Comment (mturley) [inline package.json:21]: The `--watch` param doesn't make it through to jest (this runs `npm run test:unit --watch`, which doesn't have the `--` separator). Basically when we use `-- --foo` the script we're calling can't be another `npm run`. We should remove the existing `test:watch` script and add these:

```json
"test:unit:watch": "npm run test:unit -- --watch",
"test:watch": "npm run test:unit:watch",
```

File: package.json [hunk 2]
```diff
@@ -18,10 +18,12 @@
...
     "lint": "$(npm bin)/eslint --ext .tsx,.js ./src/",
     "lint:fix": "npm run lint -- --fix",
+    "test:unit": "$(npm bin)/jest --coverage --rootDir=. --config=config/jest.config.js",
+    "test": "npm run test:unit",
+    "eslint": "$(npm bin)/eslint --ext .tsx,.js ./src/",
```
Comment (mturley) [inline package.json:26]: Do we need both the `lint` script and the `eslint` one? we should just pick one and drop the other, imo. I like the consistency of naming the scripts after what they're for, not the tool (so `type-check`, `lint` and `test`, not `tsc`, `eslint` and `jest`).

Comment (gildub) [reply package.json:26]: No we don't need both.  I agree with naming.

File: package.json (hunk 1 above)
Comment (gildub) [reply package.json:21]: Thanks for highlighting this.
I think we just need this:
```
"test": "npm run test:unit -- --coverage",
"test:unit": "$(npm bin)/jest --rootDir=. --config=config/jest.config.js",
"test:watch": "npm run test:unit -- --watch",
```

**Pull request merged**